### DISCUSSION TITLE
HA intergration to PV Output - Sensor question.

### DISCUSSION THREAD

**Discussion by DerRonnyUK** [nathanmarlor/foxess_modbus#652] [Q&A] [answered]
I am trying to get this installed and configured.

but I am struggling with the sensor names, seems they are different in Nathans Modbus integration.

any help appreicated !

struggling mostly with this part.
```

template:
  - sensor:
    # pvoutput.org upload using CSV format
    - name: "PVOutput Data"
      unique_id: foxess_inv1_pvoutput_data
      state: >
        {% set dat = now().year|string + ('0' + (now().month|string))[-2:] + ('0' + (now().day|string))[-2:] %}
        {% set gen = states('sensor.pv_energy_daily') | float(default=-1) * 1000 %}
        {% set gen = '' if gen < 0 else gen | int | string %}
        {% set exp = states('sensor.feedin_daily') | float(default=-1) * 1000 %}
        {% set exp = '' if exp < 0 else exp | int | string %}
        {% set con = states('sensor.load_daily') | float(default=-1) * 1000 %}
        {% set con = '' if con < 0 else con | int | string %}
        {% set imp = states('sensor.grid_daily') | float(default=-1) * 1000 %}
        {% set imp = '' if imp < 0 else imp | int | string %}
        {% set result = dat + ',' + gen + ',' + exp + ',,,,,,,0,' + imp + ',0,0,' + con + ',,,,' %}
        {{ result }}
```

as well this is a bit unclear, or not possible for me - I don`t have these 3 dots.

Go to Settings, Automations and CREATE a new AUTOMATION, click the three dots and select Edit in YAML and paste the following code, replacing the default yaml:
```
  alias: Upload pvoutput data every hour
  trigger:
  - platform: time_pattern
    minutes: '59'
  action:
  - service: rest_command.pvoutput_upload
  mode: single
```

**Comment by FozzieUK** [comment, marked as answer]
Assuming you haven’t set an inverter name the sensor names will be:-

sensor.pv_energy_daily = sensor.solar_energy_today

sensor.feedin_daily = sensor.feed_in_energy_today

sensor.load_daily = sensor.load_energy_today

sensor.grid_daily = sensor.grid_consumption_energy_today

.
On the second question when you create an automation in the top right hand corner of the screen will be the 3 dots 

![IMG_1640](https://github.com/user-attachments/assets/d6ca3460-f08d-406d-bc65-8418322c5d23)

**Reply by DerRonnyUK** [reply]
Inverter name? not that I am aware off ! 
thanks for the sensor names and the screenshot - being blind today.

**Reply by DerRonnyUK** [reply]
would you have any idea where I should check if the "traces" are empty and no upload to PVoutput is happening?

**Reply by FozzieUK** [reply]
Sorry I wouldn’t myself, but i’m sure that @TonyM1958 will be able to help.

**Reply by DerRonnyUK** [reply]
thanks I will wait for Tony to reply ! :-)

**Reply by TonyM1958** [reply]
Hi.

First thing to check is that the template is working. Go to Settings, Devices, Entities and search for pvout. Click on the PVOuput Data and you should see something like this:

![image](https://github.com/user-attachments/assets/67aef263-edf7-475b-adfb-a9199fdeec2e)

If it is unavailable or doesn't look similar to this, there is a problem with the template the selects and formats the sensors for the upload.

**Reply by TonyM1958** [reply]
Next thing to check is that you have setup rest_command and configured this. It should look like this:

![image](https://github.com/user-attachments/assets/a2e9a6b4-3cf6-48a7-b97a-63d7408103db)

This uses your pvoutput API Key and your systemid. These needs to be included in the above command or added to secrets.yaml.

**Reply by TonyM1958** [reply]
FInally, check the automation is configured correctly:

![image](https://github.com/user-attachments/assets/1c20cd23-c7b8-4437-b47a-9cae986efc12)

If you manually Run the automation and go to Traces, you should see a timeline like this:

![image](https://github.com/user-attachments/assets/d2e84e04-04d8-4376-8ab9-f5feb1b560f1)

**Reply by TonyM1958** [reply]
Alternative sensor names are covered in the setup wiki: 

![image](https://github.com/user-attachments/assets/9988c42f-3ab3-4fa1-b53f-0b36873ed972)


Make sure you have added rest_command to your 'configuration.yaml' and you have restarted HA so it has read this line and loaded the code needed.

**Reply by DerRonnyUK** [reply]
> Hi.
> 
> First thing to check is that the template is working. Go to Settings, Devices, Entities and search for pvout. Click on the PVOuput Data and you should see something like this:
> 
> ![image](https://private-user-images.githubusercontent.com/63789168/350710881-67aef263-edf7-475b-adfb-a9199fdeec2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE1MTE3MzYsIm5iZiI6MTcyMTUxMTQzNiwicGF0aCI6Ii82Mzc4OTE2OC8zNTA3MTA4ODEtNjdhZWYyNjMtZWRmNy00NzViLWFkZmItYTkxOTlmZGVlYzJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDIxMzcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNTkwYzFmYmMwY2UxYmYyODE1YTk0MWYyNDQ3ZWE4ZmZhY2NlYjczNDcyNWI3YTM5NzVkMjllZjk2YzAxNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gpJqOTgiPmLwkEJPtR1Dhe_K6RynXsStUKf3lVXj0lk)
> 
> If it is unavailable or doesn't look similar to this, there is a problem with the template the selects and formats the sensors for the upload.

mine looks like this

![image](https://github.com/user-attachments/assets/31fc7974-bde8-4282-b33c-3bf5dbc6f868)

**Reply by DerRonnyUK** [reply]
> Next thing to check is that you have setup rest_command and configured this. It should look like this:
> 
> ![image](https://private-user-images.githubusercontent.com/63789168/350711075-a2e9a6b4-3cf6-48a7-b97a-63d7408103db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE1MTE3OTYsIm5iZiI6MTcyMTUxMTQ5NiwicGF0aCI6Ii82Mzc4OTE2OC8zNTA3MTEwNzUtYTJlOWE2YjQtM2NmNi00OGE3LWI5N2EtNjNkNzQwODEwM2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDIxMzgxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmNzgyZTg1MzQ4MmEzM2M5NjVhY2E0ODdjZmI3ODVlNjM4ODdkMjI5MTA0MTAyMGJjNDAzZDAzMjA4NmEzMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tsApvLEt55WK4QQ_c3OzzMIN30Oxg_OpAGiAcKPhyw8)
> 
> This uses your pvoutput API Key and your systemid. These needs to be included in the above command or added to secrets.yaml.

I have done this as well - separately having the API and System ID in the secrets as well as included in this directly.

**Reply by DerRonnyUK** [reply]
> it is unavailable or doesn't look similar to this, there is a problem with the template the selects and formats the sensors for the upload.

right - here was an error - now it shows entries in the trace as well.

**Reply by DerRonnyUK** [reply]
> Alternative sensor names are covered in the setup wiki:
> 
> ![image](https://private-user-images.githubusercontent.com/63789168/350711435-9988c42f-3ab3-4fa1-b53f-0b36873ed972.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE1MTI3NTUsIm5iZiI6MTcyMTUxMjQ1NSwicGF0aCI6Ii82Mzc4OTE2OC8zNTA3MTE0MzUtOTk4OGM0MmYtM2FiMy00ZmExLWI1M2YtMGIzNjg3M2VkOTcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDIxNTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzYmVkNDQ0NDQzNWYwMTRiNmNkNjhlOGM1NTcwOWE5ODBhMDc0MGU1YmIxYWU2YzY1ZDVhNDUxNDZjM2RkMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.y1m4A8cNYi0VVyx92LhzDqgVAmPaxLjuWu2bectGW5A)
> 
> Make sure you have added rest_command to your 'configuration.yaml' and you have restarted HA so it has read this line and loaded the code needed.

that how the templates yaml looks like. 
![image](https://github.com/user-attachments/assets/8f412e66-b648-4b7f-8baf-92dd118f648e)

**Reply by DerRonnyUK** [reply]
@TonyM1958  I checked everything again - any further ideas?

**Reply by TonyM1958** [reply]
Looking at what you've posted, I don't think your template is working. The sensor you have shown is "PVOutput" when you should be looking for "PVOutput Data" that looks like the screen shot I posted (it contains CSV data and updates each time your sensors update).

Most likely, this is because you've added a new 'template.yaml' file but have not included this in your 'configuration.yaml'. You need this line, with no indent, in 'configuration.yaml':

```
template: !include template.yaml
```

As you have created a new 'template.yaml' rather than adding to an existing one, you need this line, with no indent, at the start of your 'template.yaml' file:

```
- sensor:
```

Once you make these changes, check the configuration works and restart HA and check the "PVOutput Data" entity is present and is showing values similar to my screen shot.

After you run the automation, you can also check the file ' home-assistant.log' in your CONFIG folder so see if any errors are being flagged.

**Reply by DerRonnyUK** [reply]
@TonyM1958  thanks for your input - I did not receive a notification for your post - but managed it beforehand. as you said - there was something missing but dunno what exactly, I may double check later the current status with my sceenshots.
